### PR TITLE
Animations: full on-action API

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #9129.

### CODE REVIEW

File: extensions/amp-animation/0.1/amp-animation.js [hunk 1]
```diff
@@ -215,10 +291,11 @@ export class AmpAnimation extends AMP.BaseElement {
   }
 
   /**
+   * @param {?JSONType=} opt_args
```
Comment (alanorozco) [inline extensions/amp-animation/0.1/amp-animation.js]: Might be good to consider stronger typing for "args".

File: extensions/amp-animation/0.1/web-animations.js [hunk 2]
```diff
@@ -188,6 +188,22 @@ export class WebAnimationRunner {
   }
 
   /**
+   * @param {time} time
```
Comment (alanorozco) [inline extensions/amp-animation/0.1/web-animations.js]: is `time` a default closure type?

File: extensions/amp-animation/0.1/web-animations.js [hunk 3]
```diff
@@ -476,12 +496,19 @@ export class MeasureScanner extends Scanner {
...
   /**
    * @param {!WebAnimationDef|!Array<!WebAnimationDef>} spec
+   * @param {?JSONType=} opt_args
    * @return {!Promise<!Array<!InternalWebAnimationRequestDef>>}
    */
-  resolveRequests(spec) {
-    this.css_.withVars(this.vars_, () => {
-      this.scan(spec);
-    });
+  resolveRequests(spec, opt_args) {
```
Comment (alanorozco) [inline extensions/amp-animation/0.1/web-animations.js]: nit: might be a good idea to include a high-level overview of request resolution and animation lifecycle.

File: third_party/closure-compiler/externs/web_animations.js [hunk 4]
```diff
@@ -27,6 +27,12 @@
 class Animation {
 
   /**
+   * Current time of the animation.
+   * @type {number}
```
Comment (alanorozco) [inline third_party/closure-compiler/externs/web_animations.js]: `number` doesn't match `type` as set previously

File: extensions/amp-animation/0.1/web-animations.js (hunk 2 above)
Comment (dvoytenko) [inline extensions/amp-animation/0.1/web-animations.js]: No. It's externed in AMP specifically.

File: third_party/closure-compiler/externs/web_animations.js (hunk 4 above)
Comment (dvoytenko) [inline third_party/closure-compiler/externs/web_animations.js]: Yeah. We can't use `time` here, b/c it's also an extern. Even harder, this extern has to go to the Closure's native externs at some point. So, on this level, we can't use `time`. But in our own code, we can.

File: extensions/amp-animation/0.1/web-animations.js (hunk 3 above)
Comment (dvoytenko) [inline extensions/amp-animation/0.1/web-animations.js]: Done

File: extensions/amp-animation/0.1/amp-animation.js (hunk 1 above)
Comment (dvoytenko) [inline extensions/amp-animation/0.1/amp-animation.js]: Done

**Comment by rsimha-amp** [comment]
@dvoytenko, looks like this PR broke integration tests. See https://travis-ci.org/ampproject/amphtml/jobs/238135187

**Comment by dvoytenko** [comment]
All travis tests passed. But otherwise - I don't really understand that error. @erwinmombay do you know what this is?

**Comment by dvoytenko** [comment]
I think I found it. Fix is here: https://github.com/ampproject/amphtml/pull/9657

**Comment by rsimha-amp** [comment]
Unfortunately, the integration tests only run on master, and are not run for PRs. That's why the error didn't shows up before submitting. I'll try to rectify this if possible.

**Pull request merged**